File: WoodPeckerDemo/WoodPeckerDemo/Modules/UserDefaults/UserDefaultsViewController.m
//
//  UserDefaultsViewController.m
//  WoodPeckerDemo
//
//  Created by xiaogang zhang on 2018/3/17.
//  Copyright © 2018年 lifebetter. All rights reserved.
//

#import "UserDefaultsViewController.h"

@interface UserDefaultsViewController ()

@end

@implementation UserDefaultsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (IBAction)addButtonPressed:(id)sender {
    NSUserDefaults * userDefaults = [NSUserDefaults standardUserDefaults];
    NSString * key = [NSString stringWithFormat:@"key%d",arc4random()];
    NSString * value = [NSString stringWithFormat:@"value%d",arc4random()];
    [userDefaults setObject:value forKey:key];
    [userDefaults synchronize];
}

@end

File: WoodpeckerMacDemo/WoodpeckerMacDemo/Pages/UserDefaults/UserDefaultsViewController.swift
//
//  UserDefaultsViewController.swift
//  WoodpeckerMacDemo
//
//  Created by xiaogang zhang on 2019/6/1.
//  Copyright © 2019 LIFE BETTER. All rights reserved.
//

import Cocoa

class UserDefaultsViewController: NSViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
    }
    
    @IBAction func addButtonClicked(_ sender: Any) {
        UserDefaults.standard.set(Date.init(), forKey: "time")
        UserDefaults.standard.synchronize()
    }
    
}
